crates/bevy_pbr/src/render/double_buffer.rs

```diff
@@ -0,0 +1,62 @@
+use std::mem;
+
+use bevy_ecs::prelude::Entity;
+use bevy_render::render_resource::{BufferUsages, BufferVec};
+use bevy_utils::EntityHashMap;
+use bytemuck::Pod;
+
+/// A double buffer of `T`.
+///
+/// Use [`DoubleBuffer::swap`] to swap buffer,
+/// access the current buffer with [`DoubleBuffer::current`],
+/// and the previous one with [`DoubleBuffer::previous`].
+#[derive(Default)]
+pub struct DoubleBuffer<T> {
```

Comment:
Not a big deal for now, but I'd like to have some `MultiBuffer<const N: usize, T>` where `N` is the number of buffers that get rotated through. It is apparently common to use 3-4 buffers of dynamic per-frame data like mesh uniforms. This is because the GPU can have 2-3 frames in flight, meaning the rendering depends on the contents of those buffers, and if we want to update the contents, wgpu would have to wait until the target buffer is no longer in use by the GPU to be able to transfer the data to it. We can take that in a separate PR though, and just make `DoubleBuffer<T>` into `MultiBuffer<2, T>`.

Reply:
That sounds like a fun thing to implement. But I think it should be part of a different PR.